File: apps/synapse-api/synapse_api/domain/entities/sales.py
from dataclasses import dataclass, field
from datetime import date

from .common import Money, OrderStatus, Region


@dataclass(frozen=True)
class SalesItem:
    product_name: str
    quantity: int
    unit_price: Money
    total: Money


@dataclass(frozen=True)
class SalesOrder:
    id: str
    customer_name: str
    region: Region
    status: OrderStatus
    date: date
    items: list[SalesItem] = field(default_factory=list)
    total: Money = field(default_factory=lambda: Money(amount=0, currency="INR"))

File: apps/synapse-web/src/domain/entities/sales.ts
import type { OrderStatus, Region } from "./common";

export interface SalesItem {
  product_name: string;
  quantity: number;
  unit_price: number;
  currency: string;
  total: number;
}

export interface SalesOrder {
  id: string;
  customer_name: string;
  region: Region;
  status: OrderStatus;
  date: string;
  items: SalesItem[];
  total: number;
  currency: string;
}

export interface SalesOrderListResponse {
  items: SalesOrder[];
  total: number;
  page: number;
}
